
This patch fixes an oops that can occur if an adapter times
out coming operational.


---

 linux-2.6.7-rc1-bjking1/drivers/scsi/ipr.c |    1 +
 1 files changed, 1 insertion(+)

diff -puN drivers/scsi/ipr.c~ipr_ioa_reset_timeout_oops2 drivers/scsi/ipr.c
--- linux-2.6.7-rc1/drivers/scsi/ipr.c~ipr_ioa_reset_timeout_oops2	2004-05-26 17:22:17.000000000 -0500
+++ linux-2.6.7-rc1-bjking1/drivers/scsi/ipr.c	2004-05-26 17:22:36.000000000 -0500
@@ -4818,6 +4818,7 @@ static int ipr_reset_enable_ioa(struct i
 	ipr_cmd->timer.data = (unsigned long) ipr_cmd;
 	ipr_cmd->timer.expires = jiffies + IPR_OPERATIONAL_TIMEOUT;
 	ipr_cmd->timer.function = (void (*)(unsigned long))ipr_timeout;
+	ipr_cmd->done = ipr_reset_ioa_job;
 	add_timer(&ipr_cmd->timer);
 	list_add_tail(&ipr_cmd->queue, &ioa_cfg->pending_q);
 

_
